samples/client/others/csharp-netcore-complex-files/src/Org.OpenAPITools/Client/ApiClient.cs

```diff
@@ -432,54 +436,32 @@ private ApiResponse<T> ToApiResponse<T>(IRestResponse<T> response)
         private ApiResponse<T> Exec<T>(RestRequest req, RequestOptions options, IReadableConfiguration configuration)
         {
             var baseUrl = configuration.GetOperationServerUrl(options.Operation, options.OperationIndex) ?? _baseUrl;
-            RestClient client = new RestClient(baseUrl);
-
-            client.ClearHandlers();
-            var existingDeserializer = req.JsonSerializer as IDeserializer;
-            if (existingDeserializer != null)
-            {
-                client.AddHandler("application/json", () => existingDeserializer);
-                client.AddHandler("text/json", () => existingDeserializer);
-                client.AddHandler("text/x-json", () => existingDeserializer);
-                client.AddHandler("text/javascript", () => existingDeserializer);
-                client.AddHandler("*+json", () => existingDeserializer);
-            }
-            else
-            {
-                var customDeserializer = new CustomJsonCodec(SerializerSettings, configuration);
-                client.AddHandler("application/json", () => customDeserializer);
-                client.AddHandler("text/json", () => customDeserializer);
-                client.AddHandler("text/x-json", () => customDeserializer);
-                client.AddHandler("text/javascript", () => customDeserializer);
-                client.AddHandler("*+json", () => customDeserializer);
-            }
 
-            var xmlDeserializer = new XmlDeserializer();
-            client.AddHandler("application/xml", () => xmlDeserializer);
-            client.AddHandler("text/xml", () => xmlDeserializer);
-            client.AddHandler("*+xml", () => xmlDeserializer);
-            client.AddHandler("*", () => xmlDeserializer);
+            var cookies = new CookieContainer();
 
-            client.Timeout = configuration.Timeout;
-
-            if (configuration.Proxy != null)
+            if (options.Cookies != null && options.Cookies.Count > 0)
             {
-                client.Proxy = configuration.Proxy;
+                foreach (var cookie in options.Cookies)
+                {
+                    cookies.Add(new Cookie(cookie.Name, cookie.Value));
+                }
             }
 
-            if (configuration.UserAgent != null)
+            var clientOptions = new RestClientOptions(baseUrl)
             {
-                client.UserAgent = configuration.UserAgent;
-            }
+                ClientCertificates = configuration.ClientCertificates,
+                CookieContainer = cookies,
```

Comment:
@jaredtbates Out of curosity; This adds cookie inclusion logic to `Exec` the function. Any specific reason this logic also wasn't mirrored in the `ExecAsync` function? As these two now have different behavior with the same input parameters.

Reply:
Good point, I think I missed that. It should've been included in both methods. If there was a reason for that difference, I can't recall it.